src/Daemon/Daemon.cpp

```diff
@@ -490,14 +496,14 @@ namespace usbguard
     USBGUARD_LOG(Trace) << "id=" << id;
     _policy.removeRule(id);
     if (_config.hasSettingValue("RuleFile")) {
-      _policy.save(_config.getSettingValue("RuleFile"));
+      _policy.getRuleSet()->save(_config.getSettingValue("RuleFile"));
     }
   }
 
   const RuleSet Daemon::listRules(const std::string& query)
   {
     USBGUARD_LOG(Trace) << "entry: query=" << query; 
-    return _policy.getRuleSet();
+    return (*_policy.getRuleSet());
```

Comment:
Yes, this will have to be changed to a pointer type since the RuleSet will become a base class for the various RuleSet implementations (FileRuleSet, LDAPRuleSet, ...)